src/openloco/tutorial.cpp

```diff
@@ -1,20 +1,97 @@
 #include "tutorial.h"
+#include "config.h"
+#include "environment.h"
 #include "interop/interop.hpp"
+#include "localisation/string_ids.h"
+#include "scenario.h"
+#include "ui.h"
 
 using namespace openloco::interop;
 
 namespace openloco::tutorial
 {
     static loco_global<uint8_t, 0x00508F19> _state;
 
+    static loco_global<string_id, 0x009C8708> _tutorialString;
+    static loco_global<uint8_t, 0x009C870A> _tutorialNumber;
+
     tutorial_state state()
     {
         return (tutorial_state)*_state;
     }
 
+    static void getTutorialScenarioFilename();
+
+    // 0x0043C590
+    void start(int tutorialNumber)
+    {
+        if (tutorialNumber < 0 || tutorialNumber > 3)
+            return;
+
+        // NB: only used by tutorial widget drawing after.
+        _tutorialNumber = tutorialNumber;
+
+        // Ensure that we're in windowed mode, using dimensions 1024x768.
+        const auto& display = config::get_new().display;
+        const config::resolution_t tutorialResolution = { 1024, 768 };
+        if (display.mode != config::screen_mode::window || display.window_resolution != tutorialResolution)
+        {
+            if (!ui::setDisplayMode(config::screen_mode::window, tutorialResolution))
+                return;
+        }
+
+        // TODO(avgeffen) Maybe the window should not be scaled, either?
+
+        // Get the environment file for this tutorial.
+        static const environment::path_id tutorialFileIds[] = {
+            environment::path_id::tut1024_1,
+            environment::path_id::tut1024_2,
+            environment::path_id::tut1024_3,
+        };
+
+        auto fileId = tutorialFileIds[tutorialNumber];
+        auto tutPath = environment::get_path(fileId);
+
+        // TODO(avgeffen) Will this do?
+        FILE* filePointer = fopen(tutPath.c_str(), "r");
+        addr<0x009C8700, uint32_t>() = reinterpret_cast<uint32_t>(filePointer);
+        addr<0x009C86FC, uint32_t>() = reinterpret_cast<uint32_t>(filePointer);
+        addr<0x009C8704, uint32_t>() = reinterpret_cast<uint32_t>(filePointer);
```

Comment:
This will probably work just as well:
```suggestion
        addr<0x009C8700, FILE*>() = filePointer;
        addr<0x009C86FC, FILE*>() = filePointer;
        addr<0x009C8704, FILE*>() = filePointer;
```